app/views/transaction.py

```diff
@@ -0,0 +1,766 @@
+            if not transactions_type:
+                raise TransactionTypeObjectNotFound("Invalid Transaction Type")
+
+            # for credit
+            if transactions_type.transaction_type.lower() == 'credit':
+                bank_account.account_balance += data['transaction_amount']
+                db.session.add(bank_account)
+                transaction_status = "success"
+            # for debit
+            elif transactions_type.transaction_type.lower() == 'debit':
+                if bank_account.account_balance - 1000 - data['transaction_amount'] > 0:
+                    bank_account.account_balance -= data['transaction_amount']
+                    db.session.add(bank_account)
+                    transaction_status = "success"
+                else:
+                    transaction_status = "failed"
+
+            # fund transfer type - funds withdraw
+            fund_transfer_data = FundTransfer(
+                from_account=bank_account.account_number,
+                to_account=None
+            )
+
+            db.session.add(fund_transfer_data)
+            db.session.commit()
+            fund_transfer = fund_transfer_schema.dump(fund_transfer_data)
+
+            # create account transaction
+            bank_account_data = AccountTransactionDetails(
+                transaction_amount=data['transaction_amount'],
+                transaction_status=transaction_status,
+                bank_account_id=data['bank_account_id'],
+                transaction_type_id=data['transaction_type_id'],
+                fund_transfer_id=fund_transfer.get('id'),
+                fund_transfer_info=data['fund_transfer_info']
+            )
+
+            db.session.add(bank_account_data)
+            db.session.commit()
+            result = account_transaction_details_schema.dump(bank_account_data)
+            logger.info("Response for get request for account transaction details list {}".format(result))
+            response = ResponseGenerator(data=result,
+                                         message="Account transaction details added successfully",
+                                         success=True,
+                                         status=HTTPStatus.OK)
+            return response.success_response()
+
+        except BankAccountObjectNotFound as err:
+            logger.exception(err.message)
+            response = ResponseGenerator(data={},
+                                         message=err.message,
+                                         success=False,
+                                         status=HTTPStatus.NOT_FOUND)
+        except TransactionTypeObjectNotFound as err:
+            logger.exception(err.message)
+            response = ResponseGenerator(data={},
+                                         message=err.message,
+                                         success=False,
+                                         status=HTTPStatus.NOT_FOUND)
+        return response.error_response()
```

Comment:
Along with custom message we should handle `Exception` class as well so that other exceptions will be handled there.

Reply:
Made the changes as suggested.